sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/DefaultScheduler.java

```diff
@@ -804,6 +804,21 @@ public void statusUpdate(SchedulerDriver driver, Protos.TaskStatus status) {
                         && !StateStoreUtils.fetchTasksNeedingRecovery(stateStore, configStore).isEmpty()) {
                     revive();
                 }
+
+                // If the TaskStatus contains an IP Address, store it as a property in the StateStore.
```

Comment:
In which situations is the IP address expected to be present? CNI and/or native networking? Would make sense to spell this out in the comment here. Can also point out that this is needed in addition to the offer hostname (and the label we store against the TaskInfo for that) because that value is for the host system and doesn't apply to CNI.